pkg/controller/networkpolicy/validate.go

```diff
@@ -629,6 +629,18 @@ func numFieldsSetInPeer(peer crdv1alpha1.NetworkPolicyPeer) int {
 	return num
 }
 
+// numFieldsSetInAppliedTo returns the number of fields in use of a peer.
+func numFieldsSetInAppliedTo(peer crdv1alpha1.AppliedTo) int {
+	num := 0
+	v := reflect.ValueOf(peer)
+	for i := 0; i < v.NumField(); i++ {
+		if !v.Field(i).IsZero() {
+			num++
+		}
+	}
+	return num
+}
```

Comment:
Since this function is not specific to a concrete type, we could make `numFieldsSetInPeer` generic:
```
func numFieldsSetInStruct(obj interface{}) int {
	num := 0
	v := reflect.ValueOf(peer)
	for i := 0; i < v.NumField(); i++ {
		if !v.Field(i).IsZero() {
			num++
		}
	}
	return num
}
```

Reply:
Done